core/src/main/java/com/graphhopper/routing/util/AbstractFlagEncoder.java

```diff
@@ -209,16 +210,14 @@ public double getMaxSpeed() {
     /**
      * @return {@link Double#NaN} if no maxspeed found
      */
-    protected double getMaxSpeed(ReaderWay way) {
-        double maxSpeed = OSMValueExtractor.stringToKmh(way.getTag("maxspeed"));
-        double fwdSpeed = OSMValueExtractor.stringToKmh(way.getTag("maxspeed:forward"));
-        if (isValidSpeed(fwdSpeed) && (!isValidSpeed(maxSpeed) || fwdSpeed < maxSpeed))
-            maxSpeed = fwdSpeed;
-
-        double backSpeed = OSMValueExtractor.stringToKmh(way.getTag("maxspeed:backward"));
-        if (isValidSpeed(backSpeed) && (!isValidSpeed(maxSpeed) || backSpeed < maxSpeed))
-            maxSpeed = backSpeed;
-
+    protected double getMaxSpeed(IntsRef edgeFlags) {
+        double fwdSpeed = maxSpeedEnc.getDecimal(false, edgeFlags);
+        double bwdSpeed = maxSpeedEnc.getDecimal(true, edgeFlags);
+        
+        double maxSpeed = Math.min(fwdSpeed, bwdSpeed);
+        if (maxSpeed == MaxSpeed.UNSET_SPEED) {
+            return Double.NaN;
```

Comment:
Could we refactor the code so that it always use MaxSpeed.UNSET_SPEED instead of two "magic" values?

Reply:
A bit unsure about it as `MaxSpeed.UNSET_SPEED` is rather specific to our encoding and the rest of the logic rolls with `NaN` which is IMHO a better representation of an unknown value. I'd rather see the default value of the MaxSpeed changed to `NaN`.